[![Open in Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/drbob-richardson/stat220/blob/main/Assignments/Stat_220_HW2.ipynb)

**Problem 1** A baseball player has a batting average of .240 against right handed pitchers and .375 against left handed pitchers. A batting average can be thought of as the probability they will get a hit.

Part a. What is the probability the baseball player gets 3 out of 4 hits against a right handed pitcher?

In [3]:
import numpy as np
import scipy.stats as stats
import matplotlib.pyplot as plt
import pandas as pd
from scipy.stats import binom, geom

n = 4
p = 0.240
binom_prob_success_3 = binom.pmf(3, n, p)
print(f"Probability of getting 3 out of 4 hits against a right handed pitcher: {binom_prob_success_3}")


Probability of getting 3 out of 4 hits against a right handed pitcher: 0.04202496000000001


Part b. What is the probability the baseball gets 3 out of 4 hits against a left handed pitcher.

In [4]:
pl = 0.375
binom_prob_success_3_pl = binom.pmf(3, n, pl)
print(f"Probability of getting 3 out of 4 hits against a left handed pitcher: {binom_prob_success_3_pl}")

Probability of getting 3 out of 4 hits against a left handed pitcher: 0.13183593750000003


Part c. The baseball player gets 3 out of 4 hits against a randomly chosen pitcher. Based on these results, what would you assume about what hand the pitcher uses?

I would assume that the pitcher was left handed, because the player had a 13.2% chance of these results with a lefty but only 4.2% with a right-handed pitcher.

Part d. Assume that the pitcher is randomly chosen from the team where 80% of the pitchers are right handed. Use Bayes rule to determine the probability the pitcher is right handed given that the batter hits 3 out of 4 pitches.

In [9]:
p_right_handed = 0.8
p_right_given_hit = (binom_prob_success_3 * p_right_handed) / (binom_prob_success_3 * p_right_handed + binom_prob_success_3_pl * (1 - p_right_handed))
print(f"Probability the pitcher is right handed given a hit: {p_right_given_hit}")


Probability the pitcher is right handed given a hit: 0.5604527789286492


**Problem 2**  
In Arizona, the long-run probability of rain on any given day of the year is about 5%. A researcher observes that it rained three days in a row and calculates the probability of this event as follows:

$$
Pr(\text{3 days of rain}) = Pr(\text{rain on a day})^3 = 0.05^3 = 0.000125
$$

He concludes that this was an extremely rare event.  

You are asked to critique his calculation.  

**Question:** What incorrect assumption about the *relationship between consecutive days of rain* did the researcher make in this calculation, and why does that make his probability estimate misleading?  



**Problem 3** A stock trader is examining stocks. Let event A be the event that Stock A falls. Let event B be the event that Stock B falls. Assume the following.

*   Pr(A) = 0.4
*   Pr(B) = 0.25
*   Pr(A and B) = 0.15


Part a. What is the probability that either events A or B happen?

Part b. From this, can you make statement as to whether or not events A and B are mutually exclusive?

Part c. What can you say about if events A and B are independent?

**Problem 4** Each time a person shops at a grocery store, the event of catching a cold or some other virus from another shopper is independent from visit to visit and has a constant probability over the year, equal to 0.01.

Part a. In 100 trips to this store over the course of a year, the probability of catching a virus while shopping there is 100(0.01) = 1.0. What is wrong with this reasoning?

Part b. Find the correct probability in (a).

Part c. Determine the expected number of days someone would go to a grocery store before they catch a cold or some other virus.

**Problem 5** A quiz has twelve multiple-choice questions, with four possible answers for each. A student who is completely unprepared randomly guesses the answer for each question. Let Y denote the number of correct responses

Part a. What is the name of the distribution that can be used to determine the number of questionsd guessed correctly? What are the parameters of this distribution?

Part b. Find the probability that the students gets (i) all 12, (ii) none of the questions correct.

Part c. What is the expected value and variance of the number of questions guessed correctly?